addons/knowledge/models/knowledge_article.py

```diff
@@ -739,7 +741,7 @@ def _set_member_permission(self, member_id, permission):
         if not self.user_can_write:
             return False
         member = self.article_member_ids.filtered(lambda member: member.id == member_id)
-        return member.write({'permission': permission})
+        return member.sudo().write({'permission': permission})
```

Comment:
why sudo is needed here ? We check the user_can_write just before. If user can write, no sudo needed a priori

Reply:
Because user doesn't have write access on `knowledge.article.member`?